tradefed/__init__.py

```diff
@@ -5,20 +5,34 @@
 import xml.etree.ElementTree as ET
 from io import BytesIO
 from squad.plugins import Plugin as BasePlugin
+from squad.core.models import Attachment
 
 
 logger = logging.getLogger()
 
 
+class ExtractedResult(object):
+    contents = None
+    length = None
+
+
+class ResultFiles(object):
+    test_results = None
+    tradefed_logcat = None
+    tradefed_stdout = None
+
+
 class Tradefed(BasePlugin):
     name = "Tradefed"
+    tradefed_results_url = None
 
     def __assign_test_log(self, buf, test_list):
         if buf is None:
             logger.warning("Results file doesn't exist")
             return
         # assume buf is a file-like object
         tradefed_tree = ET.parse(buf)
+        buf.seek(0)
```

Comment:
`buf` is not used afterwards, so this is probably not needed

Reply:
It is used in line 172 and than 140. That's why seek(0) is here. It wasn't needed before as this file wasn't directly attached to TestRun as Attachment.

Reply:
as right I missed the fact that `buf` was being passed in as an argument